sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -92,20 +98,22 @@ private void submitOpenConnectionTaskOutsideLoopInternal(OpenConnectionTask open
             return;
         }
 
-        this.endpointsUnderMonitorMap.compute(addressUriAsString, (key, taskList) -> {
-            if (taskList == null) {
-                taskList = new ArrayList<>();
-            }
+        synchronized (this.endpointsUnderMonitorMapReadLock) {
```

Comment:
Same here, you use `synchronized` instead of `lock()` and `unlock()`.

Reply:
good catch, meant to use lock/unlock, but forget to change

Reply:
updated